src/main/java/org/apache/sysml/runtime/instructions/gpu/context/GPUContext.java

```diff
@@ -36,24 +36,21 @@ protected GPUContext() { }
 	
 	public abstract long getAvailableMemory();
 	
-	// Creation / Destruction of GPUContext and related handles
+	/**
+	 * Creation / Destruction of GPUContext and related handles
+	 * @return
+	 */
 	public static GPUContext createGPUContext() {
 		if(currContext == null && DMLScript.USE_ACCELERATOR) {
-			// TODO: Handle this thread and resolve concurrency related bugs if any
-			new Thread(new Runnable() {
-				@Override
-				public void run() {
-					// Lazy GPU context creation
-					synchronized(isGPUContextCreated) {
-						currContext = new JCudaContext();
-						OptimizerUtils.GPU_MEMORY_BUDGET = ((JCudaContext)currContext).getAvailableMemory();
-						isGPUContextCreated = true;
-					}
-				}
-			}).start();
+			synchronized(isGPUContextCreated) {
```

Comment:
Won't this make createGPUContext a blocking call and add additional 5 seconds in the execution time ?